src/petals/client/inference_session.py

```diff
@@ -84,7 +84,8 @@ async def _read_inputs_from_queue(queue: asyncio.Queue, input_timeout: Optional[
                 break  # this message means "done sending"
 
     def step(
-        self, inputs: torch.Tensor, prompts: torch.Tensor, hypo_ids: torch.LongTensor, *, step_id: str
+        self, inputs: torch.Tensor, prompts: torch.Tensor, hypo_ids: torch.LongTensor, *,
+        step_id: str, last_validated_position: int
```

Comment:
General suggestion: the position can be used not just for speculation, but also for other use cases, such as "regenerate response" option in chat.

Can you please rename last_validated_position (pr-wide) into something neutral? For instance, `start_position`, as in "the index of the first item on inputs in history"? Or anything else you find preferable.

If you insist on `last_validated_position`, please explain your motivation and ignore my comment.